app/models/picture_set.rb

```diff
@@ -85,13 +66,59 @@ def capture_job(num, date, dir, angle)
     def convert_thread(num, date, dir, angle)
       caption = OPTS.image_caption || date
       t = Thread.new do
-        Syscall.execute("time convert -caption '#{caption}' #{date}_#{num}.jpg -sample 600 -bordercolor Snow " \
-        "-density 100 -gravity center -pointsize #{OPTS.image_fontsize} " \
-        "-polaroid -#{angle} #{date}_#{num}#{POLAROID_SUFFIX}", dir: dir)
+        Syscall.execute("time convert -caption '#{caption}' #{date}_#{num}.jpg " \
+                                     '-sample 600 ' \
+                                     '-bordercolor Snow ' \
+                                     '-density 100' \
+                                     '-gravity center' \
+                                     "-pointsize #{OPTS.image_fontsize} " \
+                                     "-polaroid -#{angle} " \
+                                     "#{date}_#{num}#{POLAROID_SUFFIX}", dir: dir)
       end
       t.abort_on_exception = true
       t
     end
+
+  end
+
+  def initialize(date: nil)
+    @date = date
+    @path = "picture_sets/#{date}"
+    @dir = "#{PICTURE_PATH}/#{date}"
+    @animation = "#{date}#{ANIMATION_SUFFIX}"
+    @combined = "#{date}#{COMBINED_SUFFIX}"
+    @pictures = (1..4).map { |i| { polaroid: "#{date}_#{i}#{POLAROID_SUFFIX}", full: "#{date}_#{i}.jpg" } }
+  end
+
+  def destroy
+    Syscall.execute("rm -r #{date}", dir: PICTURE_PATH)
```

Comment:
Any reason why you don't use FileUtils.rm_rf?
https://ruby-doc.org/stdlib-1.9.3/libdoc/fileutils/rdoc/FileUtils.html#method-c-rm_f

Reply:
should move all file operations to FileUtils, will do.